seer_pas_sdk/core/unsupported.py

```diff
@@ -1448,3 +1448,348 @@ def _get_msdataindex_path(self, display_path: list):
             )
 
         return result
+
+    def get_search_data(
+        self, analysis_id: str, analyte_type: str, rollup: str, engine: str
+    ):
+        """
+        Get search data for a given analysis ID.
+        Args:
+            analysis_id (str): ID of the analysis.
+            analyte_type (str): Type of analyte. Must be either 'protein', 'peptide', precursor.
+            rollup (str): Rollup type. Must be either 'np' or 'panel'.
+            engine (str): Search engine. Supported engines are: raw, diann, median, median80, pepcal.
+
+        Returns:
+            pd.DataFrame: A dataframe with each row containing the following columns:
+                            'msrun_id', 'sample_id', 'nanoparticle' (if rollup is 'np'), 'protein_group', 'peptide' (for peptide),
+                            'intensity_log10', 'q_value' (for protein), 'rt' and 'irt' (for peptide)
```

Comment:
The logic of the function has to be synced with the updated set of columns

```suggestion
                            'msrun_id', 'sample_id', 'nanoparticle' (if rollup is 'np'), 'protein_group', 'peptide' (for 'peptide' and 'precursor' analyte types), 'charge' (for 'precursor' analyte type)
                            'intensity_log10', 'protein_group_q_value', 'q_value' (for 'precursor' analyte type), 'rt' and 'irt' (for 'peptide' and 'precursor' analyte type)
```